streams/cluster/src/main/java/com/hortonworks/streamline/streams/cluster/register/impl/KafkaServiceRegistrar.java

```diff
@@ -86,23 +115,68 @@ protected boolean validateServiceConfiguationsAsFlattenedMap(Map<String, String>
         return configMap.containsKey(Constants.Kafka.PROPERTY_KEY_ZOOKEEPER_CONNECT);
     }
 
-    private Component createKafkaBrokerComponent(Config config, Map<String, String> flattenConfigMap) {
-        if (!config.contains(PARAM_KAFKA_BROKER_HOSTNAMES)) {
-            throw new IllegalArgumentException("Required parameter " + PARAM_KAFKA_BROKER_HOSTNAMES + " not present.");
+    private ServiceConfiguration buildServerPropertiesServiceConfiguration(Config config) {
+        ServiceConfiguration serverProperties = new ServiceConfiguration();
+        serverProperties.setName(CONFIG_SERVER_PROPERTIES);
+
+        Map<String, String> confMap = new HashMap<>();
+
+        if (config.contains(PARAM_ZOOKEEPER_CONNECT)) {
+            String zookeeperConnect = config.getString(PARAM_ZOOKEEPER_CONNECT);
+            confMap.put(PARAM_ZOOKEEPER_CONNECT, zookeeperConnect);
+        }
+
+        if (config.contains(PARAM_SECURITY_INTER_BROKER_PROTOCOL)) {
+            String securityInterBrokerProtocol = config.getString(PARAM_SECURITY_INTER_BROKER_PROTOCOL);
+            confMap.put(PARAM_SECURITY_INTER_BROKER_PROTOCOL, securityInterBrokerProtocol);
         }
 
-        List<String> kafkaBrokerHosts;
         try {
-            kafkaBrokerHosts = config.getAny(PARAM_KAFKA_BROKER_HOSTNAMES);
-        } catch (ClassCastException e) {
-            throw new IllegalArgumentException("Required parameter " + PARAM_KAFKA_BROKER_HOSTNAMES + " should be list of string.");
+            String json = objectMapper.writeValueAsString(confMap);
+            serverProperties.setConfiguration(json);
+        } catch (JsonProcessingException e) {
+            throw new RuntimeException(e);
+        }
+        return serverProperties;
+    }
+
+    private ServiceConfiguration buildKafkaEnvServiceConfiguration(Config config) {
+        ServiceConfiguration serverProperties = new ServiceConfiguration();
+        serverProperties.setName(CONFIG_KAFKA_ENV);
+
+        Map<String, String> confMap = new HashMap<>();
+
+        try {
+            String json = objectMapper.writeValueAsString(confMap);
```

Comment:
Yes some configurations will be filled while importing from Ambari, but these are actually not needed so just putting empty JSON.